src/services/notifier/hooks/subscriptions.hook.ts

```diff
@@ -46,29 +46,25 @@ export default {
           ]
         })
 
-        // provider -> {consumer} -> [hashes]
-        const hashesByProviderAndConsumer = accountSubscriptions.reduce(
-          (acc: HashesByProviderAndConsumer, current: SubscriptionModel) => {
-            const { consumer, hash, provider: { url } } = current
+        // providerUrl -> [consumerAddress]
+        const consumersByProviders = accountSubscriptions.reduce(
```

Comment:
If the tmpQuery has consumerAddress, then we don't need to group by consumers right? since it's already for a given consumer. we just need to find distinct provider url from the given set of subscriptions for the given consumer. So, what about having logic like
```

if (tmpQuery) {
  const providerList = new Set(accountSubscriptions.map(subscription=>subscription.url))
  providerList.forEach(prov=>updateSubscriptionsBy(prov, consumerAddress)
}
}
```

Also we should also consider a scenario where there are no subscriptions for a given provider in cache, but the notifier provider has some subscriptions created that are not in cache. Meaning it's a new provider from consumer perspective.

So would it be safe to do something 
```
const providers = await ProviderModel.findAll()
providers.forEach(p=>updateSubscriptionsBy(p.url, consumerAddress)
```

However this may have a performance impact if there are large number of providers.

Reply:
I agree  with the first suggestion, for sure. The second one could be expensive in the long term, though if we are the only providers for the foreseeable future then it should be fine, too. What do you think @artem-brazhnikov?

Reply:
@jurajpiar @vjgene I think we can create a separate story for this and address if it becomes a priority in the future. @vjgene document it please in jira

Reply:
@artem-brazhnikov @jurajpiar created https://rsklabs.atlassian.net/jira/software/projects/RMKT/boards/60?selectedIssue=RMKT-822

Reply:
Done with first suggestion, thanks